java/runtime-common/src/main/java/org/ray/core/RayRuntime.java

```diff
@@ -357,69 +349,53 @@ public RayObjects call(UniqueID taskId, Class<?> funcCls, RayFunc lambda, int re
-      Pair<T, GetStatus> ret = objectStoreProxy
-          .get(objectId, params.default_first_check_timeout_ms, isMetadata);
-
-      wasBlocked = (ret.getRight() != GetStatus.SUCCESS);
-
-      // Try reconstructing the object. Try to get it until at least PlasmaLink.GET_TIMEOUT_MS
-      // milliseconds passes, then repeat.
-      while (ret.getRight() != GetStatus.SUCCESS) {
-        RayLog.core.warn(
-            "Task " + taskId + " Object " + objectId.toString() + " get failed, reconstruct ...");
-        // Do another fetch
-        localSchedulerProxy.reconstructObject(objectId, false);
-
-        if (!params.use_raylet) {
-          objectStoreProxy.fetch(objectId);
-        }
-
-        //Check the result every 5s, but it will return once available.
-        ret = objectStoreProxy.get(objectId, params.default_get_check_interval_ms,
-            isMetadata);
-      }
-      RayLog.core.warn(
-          "Task " + taskId + " Object " + objectId.toString() + " get" + ", the result " + ret
-          .getLeft());
-      return ret.getLeft();
-    } catch (TaskExecutionException e) {
-      RayLog.core
-          .error("Task " + taskId + " Object " + objectId.toString() + " get with Exception", e);
-      throw e;
-    } finally {
-      // If the object was not able to get locally, let the local scheduler
-      // know that we're now unblocked.
-      if (wasBlocked) {
-        localSchedulerProxy.notifyUnblocked();
-      }
-    }
-
+    assert results.size() == 1;
+    return results.get(0);
   }
 
   // We divide the fetch into smaller fetches so as to not block the manager
   // for a prolonged period of time in a single call.
   private void dividedFetch(List<UniqueID> objectIds) {
-    int fetchSize = objectStoreProxy.getFetchSize();
-
+    int fetchSize = params.worker_fetch_request_size;
     int numObjectIds = objectIds.size();
     for (int i = 0; i < numObjectIds; i += fetchSize) {
       int endIndex = i + fetchSize;
-      if (endIndex < numObjectIds) {
-        objectStoreProxy.fetch(objectIds.subList(i, endIndex));
+
+      List<UniqueID> fetchIds = (endIndex < numObjectIds)
+          ? objectIds.subList(i, endIndex)
+          : objectIds.subList(i, numObjectIds);
+
+      if (params.use_raylet) {
+        localSchedulerProxy.reconstructObjects(fetchIds, true);
       } else {
-        objectStoreProxy.fetch(objectIds.subList(i, numObjectIds));
+        objectStoreProxy.fetch(fetchIds);
+      }
+    }
+  }
+
+  private void dividedReconstruct(List<UniqueID> objectIds) {
+    int reconstructSize = params.worker_fetch_request_size;
+    int numObjectIds = objectIds.size();
+
+    for (int i = 0; i < numObjectIds; i += reconstructSize) {
+      int endIndex = i + reconstructSize;
+
+      List<UniqueID> reconstructIds = (endIndex < numObjectIds)
+          ? objectIds.subList(i, endIndex)
+          : objectIds.subList(i, numObjectIds);
+
+      localSchedulerProxy.reconstructObjects(reconstructIds, false);
```

Comment:
In legacy code, `reconstructObjects` can only handle one ID at a time, we should use a loop here.